app/helpers/audits_helper.rb

```diff
@@ -15,8 +15,17 @@ def id_to_label(name, change, truncate = true)
           label = name.classify.gsub('Id', '').constantize.find(change).to_label
         rescue NameError
           # fallback to the value only instead of N/A that is in generic rescue below
-          return change.to_s
+          return _("Missing(ID: %s)") % change
         end
+      when /.*_ids$/
+        existing = name.sub('_ids', '').classify.constantize.where(id: change).index_by(&:id)
+        label = change.map do |id|
+          if existing.has_key?(id)
+            existing[id].to_label
+          else
+            _("Missing(ID: %s)") % id
+          end
+        end.to_sentence
```

Comment:
I think we preferred join(', ') for listing affected taxonomies, we should probably be consistent here